packages/client/src/createClient.ts

```diff
@@ -108,11 +119,13 @@ export function createClientInternal(
     T extends
       | ObjectOrInterfaceDefinition
       | ActionDefinition<any>
-      | QueryDefinition<any>,
+      | QueryDefinition<any>
+      | Experiment<"2.0.7">,
   >(o: T): T extends ObjectTypeDefinition ? ObjectSet<T>
     : T extends InterfaceDefinition ? MinimalObjectSet<T>
     : T extends ActionDefinition<any> ? ActionSignatureFromDef<T>
     : T extends QueryDefinition<any> ? QuerySignatureFromDef<T>
+    : T extends Experiment<"2.0.7"> ? { invoke: ExperimentFns<T> }
```

Comment:
I don't fully understand why these are 2.0.7 and objectSet is 2.0.8

Reply:
Mistake on my part. I first did all the things "2.0.7" and then realized the types were actually going to be introduced in 2.0.8 and missed this one. Then it seems the `defineProperties` below with the `as` acted as a an any in effect and didn't catch it.

Reply:
Do we need to address this?